core/nodepool/cluster/cluster.go

```diff
@@ -64,17 +66,44 @@ func NewClusterRef(cfg *config.ProvidedConfig, awsDebug bool) *ClusterRef {
 	}
 }
 
-func NewCluster(provided *config.ProvidedConfig, opts config.StackTemplateOptions, awsDebug bool) (*Cluster, error) {
+func NewCluster(provided *config.ProvidedConfig, opts config.StackTemplateOptions, plugins []*pluginmodel.Plugin, awsDebug bool) (*Cluster, error) {
 	stackConfig, err := provided.StackConfig(opts)
 	if err != nil {
 		return nil, err
 	}
-	ref := NewClusterRef(provided, awsDebug)
+
+	clusterRef := NewClusterRef(provided, awsDebug)
+
 	c := &Cluster{
 		StackConfig: stackConfig,
-		ClusterRef:  ref,
+		ClusterRef:  clusterRef,
+	}
+
+	extras := clusterextension.NewExtrasFromPlugins(plugins, c.Plugins)
+
+	extraStack, err := extras.NodePoolStack()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load node pool stack extras from plugins: %v", err)
+	}
+	c.StackConfig.ExtraCfnResources = extraStack.Resources
+
+	extraWorker, err := extras.Worker()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load controller node extras from plugins: %v", err)
+	}
+	c.StackConfig.CustomSystemdUnits = append(c.StackConfig.CustomSystemdUnits, extraWorker.SystemdUnits...)
```

Comment:
* This should be moved and implemented as `func (c *Cluster) CustomSystemdUnits() []model.CustomSystemdUnit` while adding `extraWorker` as a field of the type `Cluster`
  * Mutating `StackConfig` here doesn't seem intuitive as it spreads the logic what should be included in `StackConfig`
  * Even better, probably `StackConfig` should be merged into `Cluster`